frame/glutton/src/weights.rs

```diff
@@ -0,0 +1,201 @@
+
+use frame_support::{traits::Get, weights::{Weight, constants::RocksDbWeight}};
+use sp_std::marker::PhantomData;
+
+/// Weight functions needed for pallet_glutton.
+pub trait WeightInfo {
+	fn waste_ref_time_iter() -> Weight;
+	fn waste_proof_size_some() -> Weight;
+	fn waste_proof_size_none() -> Weight;
+	fn read_limits() -> Weight;
+	fn on_idle_high_proof_waste() -> Weight;
+	fn on_idle_low_proof_waste() -> Weight;
+	fn empty_on_idle() -> Weight;
+}
+
+/// Weights for pallet_glutton using the Substrate node and recommended hardware.
+pub struct SubstrateWeight<T>(PhantomData<T>);
+impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
+	fn waste_ref_time_iter() -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `0`
+		//  Estimated: `0`
+		// Minimum execution time: 4_918_896 nanoseconds.
+		Weight::from_ref_time(5_231_716_000)
+	}
+	/// Storage: Glutton TrashData (r:1 w:0)
+	/// Proof: Glutton TrashData (max_values: None, max_size: Some(1036), added: 3511, mode: MaxEncodedLen)
+	fn waste_proof_size_some() -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `2581`
+		//  Estimated: `3511`
+		// Minimum execution time: 13_415 nanoseconds.
+		Weight::from_parts(15_645_000, 3511)
+			.saturating_add(T::DbWeight::get().reads(1_u64))
+	}
+	/// Storage: Glutton TrashData (r:1 w:0)
+	/// Proof: Glutton TrashData (max_values: None, max_size: Some(1036), added: 3511, mode: MaxEncodedLen)
+	fn waste_proof_size_none() -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `55`
+		//  Estimated: `3511`
+		// Minimum execution time: 2_676 nanoseconds.
+		Weight::from_parts(2_855_000, 3511)
+			.saturating_add(T::DbWeight::get().reads(1_u64))
+	}
+	/// Storage: Glutton Compute (r:1 w:0)
+	/// Proof: Glutton Compute (max_values: Some(1), max_size: Some(4), added: 499, mode: MaxEncodedLen)
+	/// Storage: Glutton Storage (r:1 w:0)
+	/// Proof: Glutton Storage (max_values: Some(1), max_size: Some(4), added: 499, mode: MaxEncodedLen)
+	fn read_limits() -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `55`
+		//  Estimated: `998`
+		// Minimum execution time: 3_435 nanoseconds.
+		Weight::from_parts(3_594_000, 998)
+			.saturating_add(T::DbWeight::get().reads(2_u64))
+	}
+	/// Storage: Glutton Storage (r:1 w:0)
+	/// Proof: Glutton Storage (max_values: Some(1), max_size: Some(4), added: 499, mode: MaxEncodedLen)
+	/// Storage: Glutton Compute (r:1 w:0)
+	/// Proof: Glutton Compute (max_values: Some(1), max_size: Some(4), added: 499, mode: MaxEncodedLen)
+	/// Storage: Glutton TrashData (r:1489 w:0)
+	/// Proof: Glutton TrashData (max_values: None, max_size: Some(1036), added: 3511, mode: MaxEncodedLen)
+	fn on_idle_high_proof_waste() -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `1690738`
+		//  Estimated: `5228877`
+		// Minimum execution time: 42_285_419 nanoseconds.
+		Weight::from_parts(44_395_200_000, 5228877)
+			.saturating_add(T::DbWeight::get().reads(1491_u64))
```

Comment:
For the high-proof size not quite:  

At least in theory we are good enough as well `44_395_200_000 + 25_000_000 * 1491 =  81.6 ms`.

Reply:
This should be good enough, right?

Reply:
No, i updated it.  
New values are: 100.27 ms and 100.8 ms which is very good imo.